docs/src/capi/reference/annotations.md

```diff
@@ -0,0 +1,12 @@
+# Annotations
+
+## Machine
+
+The following annotations can be set on CAPI `Machine` resources.
+
+| Name                                          | Description                                          | Values                       | Set by user |
+|-----------------------------------------------|------------------------------------------------------|------------------------------|-------------|
+| `v1beta2.k8sd.io/in-place-upgrade-to`         | Trigger a Kubernetes version upgrade on that machine | version e.g `v1.31.3`        | yes         |
```

Comment:
The value for this and the `release` annotation can be either one of `localPath=/full/path/to/k8s.snap`, `revision=123` or `channel=latest/edge`.